src/components/search-form/SearchForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ReactElement, useState } from 'react';
+import { SEARCH_PHRASE_DEFAULT } from '../../utils/data';
+import styles from './SearchForm.module.scss';
+import type { SearchFormProps } from './SearchForm.types';
+
+export default function SearchForm(props: SearchFormProps): ReactElement {
+    const [searchPhrase, setSearchPhrase] = useState<string>(SEARCH_PHRASE_DEFAULT);
+
+    function updateSearchPhrase(event: React.ChangeEvent<HTMLInputElement>): void {
+        setSearchPhrase(event.target.value);
+    }
+
+    function submitSearchPhrase(event: React.FormEvent<HTMLFormElement>): void {
+        event.preventDefault();
+        props.onSubmit(searchPhrase);
+    }
+
+    function resetAll(): void {
+        setSearchPhrase(SEARCH_PHRASE_DEFAULT);
+        props.onResetAll();
```

Comment:
It won't be simpler if instead of those three callbacks you will provide one?
```typescript
interface IProps {
  onSearchChange: (data: { q: string; section: string }) => void;
}
```